src/Moq/CaptureMatch.cs

```diff
@@ -12,12 +12,27 @@ namespace Moq
 	/// <typeparam name="T"></typeparam>
 	public class CaptureMatch<T> : Match<T>
 	{
+		private readonly Matcher matcher;
+
 		/// <summary>
 		/// Initializes an instance of the capture match.
 		/// </summary>
 		/// <param name="captureCallback">An action to run on captured value</param>
 		public CaptureMatch(Action<T> captureCallback)
-			: base(CreatePredicate(captureCallback), () => It.IsAny<T>())
+			: this(captureCallback, false)
+		{
+		}
+		
+		/// <summary>
+		/// Initializes an instance of the capture match.
+		/// </summary>
+		/// <param name="captureCallback">An action to run on captured value</param>
+		/// <param name="captureOnSuccessOnly">
+		/// Indicates whether <paramref name="captureCallback"/> should be invoked on parameter evaluation
+		/// or only after all parameters were successfully evaluated.
+		/// </param>
+		public CaptureMatch(Action<T> captureCallback, bool captureOnSuccessOnly)
+			: this(new Matcher(captureCallback, null, captureOnSuccessOnly))
```

Comment:
As mentioned earlier, I still think we should treat this as  bug fix; i.e. instead of adding new ctors, let's just change the behavior of the existing ones. (As you said, `Capture.*` are probably a little-used feature so we can probably afford a small breaking change.)